### PR TITLE
c2js - midi output and device select

### PR DESCRIPTION
**Pull request by dromer**
Additions:

* midi out
* midi device selection

### CODE REVIEW

**Comment by Reinissance** [comment]
we should maybe rename hv_worklet_start.js to something like midi_utils.js - unless we'll be using it for other shared code as well.

**Comment by dromer** [comment]
@chrisuehlinger hey we are wondering what the `hv_worklet_start.js` is exactly for.

Can you give us some more info? (as it was almost being removed in this PR)

File: hvcc/generators/c2js/c2js.py [hunk 1]
```diff
@@ -241,14 +250,6 @@ def compile(
                     externs=externs,
                     pool_sizes_kb=externs["memoryPoolSizesKb"]))
 
-            pre_js_path = os.path.join(out_dir, "hv_worklet_start.js")
```
Comment (dromer) [inline hvcc/generators/c2js/c2js.py:244]: better put this back until we know for sure what to do with it :)

File: hvcc/generators/c2js/template/hv_worklet.js [hunk 2]
```diff
@@ -108,15 +108,24 @@ class {{name}}_AudioLibWorklet extends AudioWorkletProcessor {
...
             payload: [UTF8ToString(sendName), _hv_msg_getFloat(msg, 0)]
           });
-        },
-        "viiii"
+        }
+      },
+      "viiii"
       );
-      _hv_setSendHook(this.heavyContext, sendHook);
+        _hv_setSendHook(this.heavyContext, sendHook);
```
Comment (dromer) [inline hvcc/generators/c2js/template/hv_worklet.js:128]: small style issue: indentation is not needed here

File: hvcc/generators/c2js/template/hv_wrapper.js [hunk 3]
```diff
@@ -54,7 +57,7 @@ AudioLibLoader.prototype.init = function(options) {
             sampleRate: this.webAudioContext.sampleRate,
             blockSize: blockSize,
             printHook: options.printHook,
-            sendHook: options.sendHook
+            sendHook: options.sendHook,
```
Comment (dromer) [inline hvcc/generators/c2js/template/hv_wrapper.js:60]: why the comma?

File: hvcc/generators/c2js/template/index.html [hunk 4]
```diff
@@ -101,70 +148,84 @@
...
+        if (midioutPort !== null) {
+          midioutPort.send(message);
+        }
+        else {
+          console.error("No MIDI output port available.");
+        }
+      }
 
-      function onFloatMessage(sendName, floatValue) {
+      function onSentMessage(sendName, message) {
```
Comment (dromer) [inline hvcc/generators/c2js/template/index.html:163]: maybe `onSendMessage` is more correct

**Comment by chrisuehlinger** [comment]
> @chrisuehlinger hey we are wondering what the `hv_worklet_start.js` is exactly for.
> 
> Can you give us some more info? (as it was almost being removed in this PR)

First off, sorry for ghosting without totally finishing this work. Life got really hectic that summer very suddenly. I'd like to come back and check in on this at some point, make sure it builds against the most recent versions of these APIs and with the most recent version of emscripten. I may have some time in December or January, though I can't commit to anything 100%.

Anyway, wrt this file, I think those are shims for some browser APIs that aren't present in the AudioWorklet and needed to be shimmed out so that other libraries wouldn't crash when loaded in an AudioWorklet. I'd recommend leaving that stuff in unless you've got a solid way to smoke test the outputs and make sure they don't crash.

**Comment by dromer** [comment]
Thank you for the reply @chrisuehlinger !

Looking forward to your involvement again in the future :)

**Comment by Reinissance** [comment]
Hm, seems to fail again because of flake8 complaining about the missing new line at the end in c2js, could it be this check was added after the last changes made in this file? As I completely restored it to the state before I touched it...

**Comment by dromer** [comment]
No, these checks have been in place since forever.

It seems you are not putting a proper newline here.

**Comment by dromer** [comment]
@Reinissance I'll see if I can try this today :)

**Pull request merged**